### DISCUSSION TITLE
Copilot stopped working in VS 2022 Professional

### DISCUSSION THREAD

**Discussion by cheema-corellian** [community/community#23539] [Copilot Conversations]
Another user in this forum reported that Copilot stopped working for them in VS 2022 Community Edition.  I am here to report that the same thing happened to me in the Professional Edition.  I have the extension installed as can be seen in the screenshot below.  There are no errors anywhere that I can see.  If I open the same solution in VSCode, no problem.  Copilot works great there on the same C# file.  I went ahead and update VS 2022 to the latest version 17.2.6.  That made no difference.

Funny thing is, it worked OK for a time.  And then abruptly stopped one day.   :(

<img width="412" alt="image" src="https://user-images.githubusercontent.com/63803292/181077094-84f031be-0afd-483a-97f6-54b43494ae5e.png">

**Comment by cheema-corellian** [comment]
Well, the Copilot extension started working again.  And I don't think I changed anything.  Strange.

On a side note.  In VSCode, the suggestions are good and many.  In Visual Studio 2022, suggestions are few and far between.  Maybe it is because I use VSCode more and it has had more of a chance to train on it?

**Reply by cmp-nct** [reply]
There is some logic in the plugins to learn from your behavior (reject and accept telemetry) but copilot is not that smart currently.
Also given that 80-95% of the code you "accept" is actually garbage you need to remove again you should be happy it does not learn from accepting ;-)
Copilot basically forgets everything about you after the completion (and any followup cycles for different ones) is done.
It's just a very unstable plugin that needs an IDE restart regularly.

**Comment by AlexAngstrom** [comment]
Having exactly the same issue

**Comment by pcorbineau** [comment]
Exactly same issue, sometimes quitting vs 2022, then restart fix the issue but its not a way to work ^^

**Comment by Bortronx** [comment]
Same issue here. It doesn't work. Everything is good but no suggestions:

GitHub Copilot 1.42.0.0 (v1.42.0.0@5f1bd5ff5)
GitHub Copilot Agent v1.42.0
GitHub Copilot is globally enabled

**Comment by ricardyn** [comment]
Same behavior with me. Few suggestions and sometimes none. It stops working sometimes with no apparent reason.

**Comment by cmp-nct** [comment]
On Visual Studio copilot becomes less stable with each completion, so if you develop for an hour you can expect it to stop working.
In the beginning you can close the file tab and open it again, scroll back to where you stopped and it will work again.
Once the automated completion stops working you can use Ctrl+Enter for the panel, then  click back into the source ("accept" does not work) and it will work for a while like that, possibly another half hour.
Then you need to restart the IDE.

Basically it's part of my copilot work-cycle on Visual Studio to restart the IDE every 30 minutes, then it works acceptably stable.

**Comment by cantonalex** [comment]
Having the same issue Apple Silicon VS Code.

**Comment by gcohen12** [comment]
Same here

**Comment by BryceBarbara** [comment]
My issues were solved by uninstalling Code Rush.

**Comment by BillyMartin1964** [comment]
Copilot agent works for about half a day, and then the agent dropdown disappears. If you ask the ai, they will say they are still in Agent mode, but then they can't seem to do any of the agent tasks.

![image](https://github.com/user-attachments/assets/32bca718-d34a-47e1-ade2-1d2c954d19ad)
